core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java

```diff
@@ -89,6 +89,14 @@ public void uploadUpscaleRecipes(Stack stack, HostGroup hostGroup, Set<HostMetad
         }
     }
 
+    public void executePreAmbariStartRecipes(Stack stack, Set<HostGroup> hostGroups) throws CloudbreakException {
+        Orchestrator orchestrator = stack.getOrchestrator();
+        if (recipesSupportedOnOrchestrator(orchestrator)
+            && (recipesFound(hostGroups) || (stack.getCluster() != null && stack.getCluster().getLdapConfig() != null))) {
```

Comment:
1, The conditions aren't ordered by weight
2, Hostgroups are part of cluster, is it possible that `stack.getCluster() != null` is unnecessary?

Reply:
1) I reordered it (and extracted)
2) It is necessary